.github/workflows/pipeline.yml

```
@@ -17,14 +17,44 @@ jobs:
       - run: yarn test
 
   build:
-    name: Build JSONSchema and TypeScript
+    name: Build JSONSchema, TypeScript and Web
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v3
         with:
           node-version: 18
-      # TODO build:web fails currently, but it'd be good to make that work when possible and then add it here.
       - run: yarn
-      - run: yarn build:jsonschema
-      - run: yarn build:ts
+      - run: yarn build
+
+  build-docker:
+    name: Build Docker Image
+    runs-on: ubuntu-latest
+    env:
+      PLATFORMS: linux/amd64
+      DOCKER_NAMESPACE: matrixdotorg
+      # Only push if this is develop, otherwise we just want to build
+      # On a PR github.ref is the target branch, so don't push for that either
+      PUSH: ${{ github.ref == 'refs/heads/master' && github.event_name != 'pull_request' }}
+
+    steps:
+      - name: Check out
+        uses: actions/checkout@v2
+
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        if: ${{ env.PUSH == 'true' }}
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: ${{ env.PUSH }}
+          tags: |
+            matrixdotorg/conf-bot:latest
```

This might need a ghcr.io/ in it? I would also suggest a sha tag so we can pin to a specific commit :)

```suggestion
            ghcr.io/matrix-org/conference-bot:latest
            ghcr.io/matrix-org/conference-bot:${{ github.sha }}
```